caramel/config.py

```diff
@@ -37,3 +37,37 @@ def __call__(self, parser, namespace, values, option_string=None):
                 " specify a path in the call like so:\n"
                 "\t caramel_initializedb /path/to/config.ini [...]"
             )
+
+
+def _get_config_value(
+    arguments: argparse.Namespace,
+    variable,
+    required=False,
+    setting_name=None,
+    settings=None,
+    env=None,
+):
+    """Returns what value to use and logs where it got it from, prefer argument >
+    env-variable > config-file"""
+    if env is None:
+        env = os.environ
+    env_var = "CARAMEL_" + variable.upper().replace("-", "_")
+
+    if setting_name is None:
+        setting_name = variable
+
+    config_value = None
```

Comment:
Logical ordering matters a bit here. I'd say:

start by declaring "config_value" ( preferrably renamed to "result" to make it clear when reading the early part)

Then do the "settings" part.  setting up the settings. 

After that do all the "env" related parts ( if env is None, and so on) as a single block, and later the  argument parsing part.

The code will read more coherently when rephrased in clear paragraphs like that